app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivityViewModel.java

```diff
@@ -39,4 +49,44 @@ public void setFrom(Place from) {
         this.from = from;
     }
 
+    public void setShuttles() {
+        shuttles = appDB.shuttleDao().getAll();
+    }
+
+    public List<Shuttle> getShuttles() {
+        String campusFrom = "";
+        String campusTo = "";
+        if (getFrom() != null && getTo() != null) {
+            if (from.getClass() == Building.class) {
+                campusFrom = ((Building) from).getCampus();
+            } else if (to.getClass() == Building.class) {
+                campusTo = ((Building) to).getCampus();
+            } else if (from.getClass() == Floor.class) {
+                campusFrom = ((Floor) from).getCampus();
+            } else if(to.getClass() == Floor.class) {
+                campusTo = ((Floor) to).getCampus();
+            }
+            if (campusFrom.compareTo(campusTo) == 0) {
+                return null;
+            }
+        }
+
+        Calendar cal = Calendar.getInstance();
+        String day = new SimpleDateFormat("EEEE", Locale.ENGLISH).format(cal.getTime());
+        SimpleDateFormat time = new SimpleDateFormat("HH:mm");
+        shuttles = appDB.shuttleDao().getScheduleByCampusAndDayAndTime(campusFrom, day, time.format(cal.getTime()));
+        return shuttles;
+    }
+
+    public String getShuttleDisplayText(List<Shuttle> shuttles) {
+        String content = "";
+        String campusTo = shuttles.get(0).getCampus().compareTo("SGW") == 0 ? "LOY" : "SGW";
+        if (shuttles == null || shuttles.size() == 0) {
+            return "No available routes using the shuttle service";
+        }
+        for (Shuttle shuttle : shuttles) {
+            content += shuttle.getCampus() + "  >   " + campusTo + ", leaves at: " + shuttle.getTime() + "\n";
+        }
```

Comment:
I would put this a separate method and call it later in here. What do you think ? It would make it easier to test these two methods later.

Reply:
but the method is simple enough I don't think it needs to be more seperated

Reply:
agree with Caren here

Reply:
You guys have a point. I am good with that too